### PR TITLE
remove CSRF user warning, soft-assert, login check

### PR DESCRIPTION
**Pull request by sravfeyn**
The purpose of `HQCsrfMiddleware` (to smooth rollout CSRF) has ended, so I am removing an unnecessary user warning and soft-assert. We can remove this middleware altogether and use Django's inbuilt, but still keeping this around for following two reasons
1. To log all csrf rejections in couchlog (This may be redundant as we can find all 403s in Couch log anyway). In terms of security this could be a proxy for malicious attackers trying CSRF exploit on HQ
2. A setting to turn CSRF on or off

@czue  If the above two doesn't sound that important, we can just remove the whole custom csrf middleware

cc @proteusvacuum

### CODE REVIEW

**Comment by millerdev** [comment]
kicked travis because it failed with no output for 10 minutes.

**Comment by millerdev** [comment]
Looks good to me. I wouldn't be opposed to "remove the whole custom csrf middleware." 👍 on ✅

File: corehq/apps/hqwebapp/middleware.py [hunk 1]
```diff
@@ -2,18 +2,13 @@
...
 logger = logging.getLogger('')
 
 
 class HQCsrfViewMiddleWare(CsrfViewMiddleware):
 
     def _reject(self, request, reason):
-        messages.error(request, _('If cookies are enabled and you still see this error, please report an issue'))
         request_url = request.path
         referring_url = request.META.get('HTTP_REFERER', 'Unknown URL')
         warning = "Request at {url} doesn't contain a csrf token. "\
```
Comment (millerdev) [inline corehq/apps/hqwebapp/middleware.py]: I think this warning message is not correct anymore. On the next line it says "Letting the request pass through for now." which I think is not true unless `CSRF_SOFT_MODE` is enabled.

**Comment by czue** [comment]
I'm also fine removing the middleware entirely. :+1: from me. Will let @millerdev merge since he has an open comment.

**Comment by sravfeyn** [comment]
Removing the view altogether. Updated ansible settings as well

**Comment by sravfeyn** [comment]
The only reason to keep this around is if we want to do session less API rollout gradually (which I am working on in my extra time). If you guys are sure there should be none using api via sessions (either via internal JS or external users), then we don't need to do gradual roll-out.

**Pull request merged**